nvtabular/loader/backend.py

```diff
@@ -504,4 +524,39 @@ def _create_tensors(self, gdf):
         return tensors
 
     def _handle_tensors(self, cats, conts, labels):
-        return cats, conts, labels
+        X = {}
+        for tensor, names in zip([cats, conts], [self.cat_names, self.cont_names]):
+            lists = {}
+            if isinstance(tensor, tuple):
+                tensor, lists = tensor
+            names = [i for i in names if i not in lists]
+
+            # break list tuples into two keys, with postfixes
+            # TODO: better choices for naming?
+            list_columns = list(lists.keys())
+            for column in list_columns:
+                values, nnzs = lists.pop(column)
+                lists[column] = values, nnzs
+
+            # now add in any scalar tensors
+            if len(names) > 1:
+                tensors = self._tensor_split(tensor, len(names), axis=1)
+                lists.update(zip(names, tensors))
+            elif len(names) == 1:
+                lists[names[0]] = tensor
+            X.update(lists)
+
+        for column_name in X:
+            if column_name in self.sparse_list:
+                if column_name not in self.sparse_max:
+                    raise ValueError(
+                        f"Did not convert {column_name} to sparse missing sparse_max entry"
+                    )
+                X[column_name] = self._to_sparse_tensor(X[column_name], column_name)
+
+        # TODO: use dict for labels as well?
+        # would require output layers to match naming
```

Comment:
I can see arguments both directions on this, but it seems reasonable to leave it as is for now.

Reply:
agreed -